dapps/marketplace/src/components/TelegramLoginButton.js

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, memo } from 'react'
+
+const TelegramLoginButton = ({ redirectURL, buttonText, className }) => {
+  let rootEl
+
+  useEffect(() => {
+    if (
+      !process.env.TELEGRAM_BOT_USERNAME ||
+      process.env.TELEGRAM_BOT_USERNAME === 'origin_protocol_test_bot'
+    ) {
+      // origin_protocol_test_bot is the default value set in dapps/marketplace/webpack.config.js
+      console.error(`Create a bot and set TELEGRAM_BOT_USERNAME env variable`)
+    }
+
+    const script = document.createElement('script')
```

Comment:
Should we verify this element doesn't already exist to prevent multiple scripts being attached if the component is re-rendered?

Reply:
I guess it shouldn't be a problem, since we are using `useEffect`. We want to render a different script element with different attributes when there is a change in `redirectURL`. 

The script file loading multiple times, when the component has unmounted and then mounted, shouldn't be a problem either. Since the script just injects an iframe, which gets removed when the component is unmounted, along with script element.